imap_processing/tests/ultra/unit/test_decom_apid_900.py

```diff
@@ -0,0 +1,118 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+from imap_processing import imap_module_directory
+
+
+@pytest.mark.parametrize(
+    "decom_test_data",
+    [
+        pytest.param(
+            {
+                "apid": 900,
+                "filename": "FM45_UltraFM45Extra_TV_Tests_"
+                "2024-01-22T0930_20240122T093008.CCSDS",
+            }
+        )
+    ],
+    indirect=True,
+)
+@pytest.mark.external_test_data
+def test_image_raw_events_decom(decom_test_data, ccsds_path_events, xtce_path):
+    """This function reads validation data and checks that decom data
+    matches validation data for image rate packet"""
+    filename = (
+        "ultra45_raw_sc_imgpriority3evnt_FM45_UltraFM45Extra_TV_Tests_"
+        "2024-01-22T0930_20240122T093008.csv"
+    )
+    priority_3_events_test_path = (
+        imap_module_directory / "tests" / "ultra" / "data" / "l0" / filename
+    )
+
+    decom_ultra = decom_test_data
+
+    df = pd.read_csv(priority_3_events_test_path, index_col="MET")
+
+    # # Check all values of each column are as expected,
+    # except for those set to fill value
+    np.testing.assert_array_equal(
+        df["SID"].values[df["SID"].values != -1],
+        decom_ultra["sid"].values[df["SID"].values != -1],
+    )
```

Comment:
As a general comment, this might be easier to read and keep track of with a loop over the variables. Rather than lots of individual checks.
```python
vars_to_compare = ["SID", "Spin", "AbortFlag"]
for var in vars_to_compare:
    good_values = df[var] != -1
    np.testing.assert_array_equal(df[var][good_values], decom_ultra[var.lower()].values[good_values])